nemo/collections/nlp/data/machine_translation/preproc_mt_data.py

```diff
@@ -0,0 +1,406 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import glob
+import json
+import os
+import pickle
+import tarfile
+import tempfile
+
+import youtokentome as yttm
+from pytorch_lightning import Trainer
+
+from nemo.collections.nlp.data.machine_translation.machine_translation_dataset import TranslationDataset
+from nemo.collections.nlp.models.machine_translation.mt_enc_dec_config import MTEncDecModelConfig
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_tokenizer
+from nemo.utils import logging
+
+
+class MTDataPreproc:
+    """Automatically preprocess machine translation.
```

Comment:
This docstring can be improved a little.

Reply:
Done.